public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -226,264 +147,85 @@ describe('LokiDatasource', () => {
-      expect(ds.runRangeQuery).toBeCalled();
+    afterAll(() => {
+      templateSrvStub.getAdhocFilters.mockReturnValue(originalAdhocFiltersMock);
     });
 
-    it('should run only metrics range query if no query type is selected in Explore', async () => {
-      const { ds, options } = setup(metricsQuery, CoreApp.Explore);
-      lastValueFrom(await ds.query(options));
-      expect(ds.runInstantQuery).not.toBeCalled();
-      expect(ds.runRangeQuery).toBeCalled();
-    });
-
-    it('should run only logs range query in Dashboard', async () => {
-      const { ds, options } = setup(logsQuery, CoreApp.Dashboard);
-      lastValueFrom(await ds.query(options));
-      expect(ds.runInstantQuery).not.toBeCalled();
-      expect(ds.runRangeQuery).toBeCalled();
-    });
-
-    it('should run only metrics range query in Dashboard', async () => {
-      const { ds, options } = setup(metricsQuery, CoreApp.Dashboard);
-      lastValueFrom(await ds.query(options));
-      expect(ds.runInstantQuery).not.toBeCalled();
-      expect(ds.runRangeQuery).toBeCalled();
-    });
-
-    it('should return dataframe data for metrics range queries', async () => {
-      const ds = createLokiDSForTests();
-      const options = getQueryOptions<LokiQuery>({
-        targets: [{ expr: metricsQuery, refId: 'B', range: true }],
-        app: CoreApp.Explore,
-      });
-
-      fetchMock.mockImplementation(() => of(testMetricsResponse));
-
-      await expect(ds.query(options)).toEmitValuesWith((received) => {
-        const result = received[0];
-        const frame = result.data[0] as DataFrame;
-
-        expect(frame.meta?.preferredVisualisationType).toBe('graph');
-        expect(frame.refId).toBe('B');
-        frame.fields.forEach((field) => {
-          const value = field.values.get(0);
-
-          if (field.type === FieldType.time) {
-            expect(value).toBe(1605715380000);
-          } else {
-            expect(value).toBe(1.1);
-          }
-        });
-      });
-    });
-
-    it('should return series data for logs range query', async () => {
-      const ds = createLokiDSForTests();
-      const options = getQueryOptions<LokiQuery>({
-        targets: [{ expr: logsQuery, refId: 'B' }],
-      });
-
-      fetchMock.mockImplementation(() => of(testLogsResponse));
-
-      await expect(ds.query(options)).toEmitValuesWith((received) => {
-        const result = received[0];
-        const dataFrame = result.data[0] as DataFrame;
-        const fieldCache = new FieldCache(dataFrame);
-
-        expect(fieldCache.getFieldByName('Line')?.values.get(0)).toBe('hello');
-        expect(dataFrame.meta?.limit).toBe(20);
-        expect(dataFrame.meta?.searchWords).toEqual(['foo']);
-      });
+    it('should not modify expression with no filters', async () => {
+      expect(ds.applyTemplateVariables(query, {}).expr).toBe(DEFAULT_EXPR);
     });
 
-    it('should return custom error message when Loki returns escaping error', async () => {
```

Comment:
We don't do custom error message for incorrectly escaped labels anymore?

Reply:
no, you are correct, we are not doing it currently in backend mode, and we should do that. i'll add a separate PR to add that, if that's ok with you.